### PR TITLE
Add support for secure devices pin

### PR DESCRIPTION
**Pull request by balloob**
Allow users to set a pin to interact with secure Google Assistant devices

Backend PR: https://github.com/home-assistant/home-assistant/pull/23223

### CODE REVIEW

**Comment by iantrich** [comment]
I didn't know this was a feature for GA. Nice!

**Comment by iantrich** [comment]
`google-ass` :rofl:

File: src/data/cloud.ts [hunk 1]
```diff
@@ -78,9 +80,9 @@ export const fetchCloudSubscriptionInfo = (hass: HomeAssistant) =>
 export const updateCloudPref = (
   hass: HomeAssistant,
   prefs: {
-    google_enabled?: boolean;
-    alexa_enabled?: boolean;
-    google_allow_unlock?: boolean;
+    google_enabled?: CloudPreferences["google_enabled"];
+    alexa_enabled?: CloudPreferences["alexa_enabled"];
```
Comment (iantrich) [inline src/data/cloud.ts]: Shouldn't these be required if their `CloudPreferences` counter-parts are required?

**Review by iantrich — APPROVED** [review]

File: src/data/cloud.ts (hunk 1 above)
Comment (balloob) [reply src/data/cloud.ts]: We only submit the preferences that we want to change when updating the prefs.

**Pull request merged**